src/eopf_geozarr/conversion/geozarr.py

```diff
@@ -1618,6 +1676,46 @@ def _add_grid_mapping_variable(
                 print(f"  Added grid_mapping attribute to {var_name}")
 
 
+def _calculate_shard_dimension(data_dim: int, chunk_dim: int) -> int:
```

Comment:
if I understand this function correctly, it doesn't set an upper limit on the shard size, is that OK?